addons/account/models/account.py

```diff
@@ -1225,132 +1251,186 @@ def write(self, vals):
         # create the relevant refund sequence
         if vals.get('refund_sequence'):
             for journal in self.filtered(lambda j: j.type in ('sale', 'purchase') and not j.refund_sequence_id):
-                journal_vals = {
+                journal.refund_sequence_id = self._create_refund_sequence({
                     'name': journal.name,
+                    'type': vals.get('type', journal.type),
                     'company_id': journal.company_id.id,
                     'code': journal.code,
                     'refund_sequence_number_next': vals.get('refund_sequence_number_next', journal.refund_sequence_number_next),
-                }
-                journal.refund_sequence_id = self.sudo()._create_sequence(journal_vals, refund=True).id
-        # Changing the 'post_at' option will post the draft payment moves and change the related invoices' state.
-        if 'post_at' in vals and vals['post_at'] != 'bank_rec':
-            draft_moves = self.env['account.move'].search([('journal_id', 'in', self.ids), ('state', '=', 'draft')])
-            pending_payments = draft_moves.mapped('line_ids.payment_id')
-            pending_payments.mapped('move_line_ids.move_id').post()
-            pending_payments.mapped('reconciled_invoice_ids').filtered(lambda x: x.state == 'in_payment').write({'state': 'paid'})
+                }, {
+                    'name': _('%s: Refund Sequence') % vals['code'],
+                    'prefix': 'R' + vals['code'].upper() + '/%(range_year)s/',
+                    'implementation': 'no_gap',
+                    'padding': 4,
+                    'number_increment': 1,
+                    'use_date_range': True,
+                    'company_id': journal.company_id.id,
+                }).id
         for record in self:
             if record.restrict_mode_hash_table and not record.secure_sequence_id:
                 record._create_secure_sequence(['secure_sequence_id'])
 
         return result
 
     @api.model
-    def _get_sequence_prefix(self, code, refund=False):
-        prefix = code.upper()
-        if refund:
-            prefix = 'R' + prefix
-        return prefix + '/%(range_year)s/'
+    def get_next_bank_cash_default_code(self, journal_type, company):
+        journal_code_base = (journal_type == 'cash' and 'CSH' or 'BNK')
+        journals = self.env['account.journal'].search([('code', 'like', journal_code_base + '%'), ('company_id', '=', company.id)])
+        for num in range(1, 100):
+            # journal_code has a maximal size of 5, hence we can enforce the boundary num < 100
```

Comment:
I don't know why this appears as a diff because this method is already there in master:
https://github.com/odoo/odoo/blob/master/addons/account/models/account.py#L1307